screens/onboarding/PermissionsScreen.js

```diff
@@ -0,0 +1,262 @@
+    await this._findCurrentLocation();
+    this.navigateStoreSelect();
+  }
+
+  _findCurrentLocation = async () => {
+    const { status } = await Permissions.askAsync(Permissions.LOCATION);
+    // Error message not checked anywhere
+    if (status !== 'granted') {
+      this.setState({
+        locationErrorMsg: 'Permission to access location was denied',
+      });
+      Analytics.setUserProperty('location_permissions', 'denied');
+    } else {
+      try {
+        const location = await Location.getCurrentPositionAsync({
+          timeout: 3000,
+        });
+        const region = {
+          latitude: location.coords.latitude - deltas.latitudeDelta / 3.5,
+          longitude: location.coords.longitude,
+          ...deltas,
+        };
+        this.setState({ locationErrorMsg: null, region });
+        Analytics.setUserProperty('location_permissions', 'granted');
+      } catch (err) {
+        console.log(err);
+        this.setState({
+          locationErrorMsg: 'Permission to access location was denied',
+        });
+        Analytics.setUserProperty('location_permissions', 'error');
+        logErrorToSentry({
+          screen: 'MapScreen',
+          function: '_findCurrentLocation',
+          error: err,
+        });
+      }
+    }
+  };
+
+  // Calculate distances and sort by closest to location
+  // _findCurrentLocation populates this.state.region with the correct lat/lon
+  // Since it's initially set to a default value, we use that instead of this.state.location
+  _orderStoresByDistance = async (stores) => {
+    const sortedStores = [];
+    // Display distance in the StoreList
+    await stores.forEach(async (store) => {
+      const currStore = store;
+      currStore.distance = findDistance(this.state.region, store);
+      const [closestZip] = await geo2zip(currStore);
+      currStore.zip = closestZip;
+      sortedStores.push(currStore);
+    });
+    // sorts in place
+    sortedStores.sort(function compare(a, b) {
+      return a.distance - b.distance;
+    });
+
+    const defaultStore = stores.find((store) => {
+      return store.id === RecordIds.defaultStoreId;
+    });
+
+    const region = {
+      latitude: defaultStore.latitude,
+      longitude: defaultStore.longitude,
+      ...deltas,
+    };
+
+    // Condition for showDefaultStore requires prevState, so a little messy
+    this.setState((prevState) => ({
+      stores: sortedStores,
+      store:
+        prevState.locationErrorMsg || sortedStores[0].distance > 100
+          ? defaultStore
+          : sortedStores[0],
+      showDefaultStore: prevState.locationErrorMsg
+        ? true
+        : sortedStores[0].distance > 100,
+      region:
+        prevState.locationErrorMsg || sortedStores[0].distance > 100
+          ? region
+          : prevState.region,
+    }));
+    Analytics.setUserProperty(
+      'closest_store',
+      this.state.showDefaultStore ? 'default' : this.state.store.storeName
+    );
+  };
+
+  updateStep = () => {
```

Comment:
I wanted to consolidate the 2 permissions screens (selecting favorite stores and enabling notifications) into this 1 screen, so just have a `step` variable in the state + this callback function to update it (since completing step 1 takes you back to this page). 
1. Is this a good way to handle it? It feels very jank
2. It makes the navigation look a bit funky because the animation after finishing the 'select stores' step looks like it's going back, though it should look like it's going forward.